### ISSUE TITLE
F811 not working in specific case

### ISSUE THREAD

**Issue by albertopoljak** [astral-sh/ruff#9305]
RUFF rule F811 does not catch this case:

```py
from test import Test


class Test(Test):
    ...
```

When tested with PyLint:
`test2.py:4:0: E0102: class already defined line 1 (function-redefined)`

Flake8 does not flag this.

**Comment by zanieb** [comment]
Hm `F811` is [`redefined-of-unused`](https://docs.astral.sh/ruff/rules/redefined-while-unused/) and your variable `Test` is being used before it is redefined, I believe this is as-designed.

**Comment by albertopoljak** [comment]
> Hm `F811` is [`redefined-of-unused`](https://docs.astral.sh/ruff/rules/redefined-while-unused/) and your variable `Test` is being used before it is redefined, I believe this is as-designed.

I found F811 to be the most similar to PyLint E0102. I also tried Ruff with ALL but to no avail.

Is there any other rule that should catch this case?

**Comment by zanieb** [comment]
I don't think this is implemented another way. Maybe we need to actually implement the Pylint rule, as I think F811 is doing what it is supposed to here but it does seem useful to raise a violation for a redefined name here.

**Comment by charliermarsh** [comment]
👍 I've unchecked E0102 from https://github.com/astral-sh/ruff/issues/970. Merging into that issue.
